go/cmd/dolt/commands/branch.go

```diff
@@ -243,159 +320,131 @@ func printAllDatasets(ctx context.Context, dEnv *env.DoltEnv) int {
-				verr = bdr.AddCause(err).Build()
-			}
-			return HandleVErrAndExitCode(verr, usage)
-		}
+	if apr.Contains(cli.RemoteParam) {
+		cli.PrintErrln("--remote/-r can only be supplied when listing or deleting branches, not when copying branches")
+		return 1
 	}
 
-	return HandleVErrAndExitCode(nil, usage)
+	return runQuery(sqlCtx, queryEngine, args)
 }
 
-func createBranch(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults, usage cli.UsagePrinter) int {
-	if apr.NArg() == 0 || apr.NArg() > 2 {
+func deleteBranches(sqlCtx *sql.Context, queryEngine cli.Queryist, apr *argparser.ArgParseResults, args []string, usage cli.UsagePrinter) int {
+	if apr.NArg() == 0 {
 		usage()
 		return 1
 	}
 
-	var remote string
-	var remoteBranch string
-	newBranch := apr.Arg(0)
-	startPt := "head"
-	if apr.NArg() == 2 {
-		startPt = apr.Arg(1)
+	if apr.Contains(cli.AllFlag) {
+		cli.PrintErrln("--all/-a can only be supplied when listing branches, not when deleting branches")
+		return 1
 	}
 
-	trackVal, setTrackUpstream := apr.GetValue(cli.TrackFlag)
-	if setTrackUpstream {
-		if trackVal == "inherit" {
-			return HandleVErrAndExitCode(errhand.BuildDError("--track='inherit' is not supported yet").Build(), usage)
-		} else if trackVal == "direct" && apr.NArg() != 2 {
-			return HandleVErrAndExitCode(errhand.BuildDError("invalid arguments").Build(), usage)
-		}
-
-		remotes, err := dEnv.RepoStateReader().GetRemotes()
-		if err != nil {
-			return HandleVErrAndExitCode(errhand.BuildDError(err.Error()).Build(), usage)
-		}
-
-		if apr.NArg() == 2 {
-			// branchName and startPt are already set
-			remote, remoteBranch = actions.ParseRemoteBranchName(startPt)
-			_, remoteOk := remotes[remote]
-			if !remoteOk {
-				return HandleVErrAndExitCode(errhand.BuildDError("'%s' is not a valid remote ref and a branch '%s' cannot be created from it", startPt, newBranch).Build(), usage)
-			}
-		} else {
-			// if track option is defined with no value,
-			// the track value can either be starting point name OR branch name
-			startPt = trackVal
-			remote, remoteBranch = actions.ParseRemoteBranchName(startPt)
-			_, remoteOk := remotes[remote]
-			if !remoteOk {
-				newBranch = trackVal
-				startPt = apr.Arg(0)
-				remote, remoteBranch = actions.ParseRemoteBranchName(startPt)
-				_, remoteOk = remotes[remote]
-				if !remoteOk {
-					return HandleVErrAndExitCode(errhand.BuildDError("'%s' is not a valid remote ref and a branch '%s' cannot be created from it", startPt, newBranch).Build(), usage)
-				}
-			}
-		}
+	if apr.Contains(cli.VerboseFlag) {
+		cli.PrintErrln("--verbose/-v can only be supplied when listing branches, not when deleting branches")
+		return 1
 	}
 
-	err := actions.CreateBranchWithStartPt(ctx, dEnv.DbData(), newBranch, startPt, apr.Contains(cli.ForceFlag), nil)
+	return runQuery(sqlCtx, queryEngine, args)
+}
+
+func runQuery(sqlCtx *sql.Context, queryEngine cli.Queryist, args []string) int {
```

Comment:
Rename this function, and add a comment about this. copyBranch, createBranch, deleteBranches, and moveBranch all come here - after they validate arguments. But other commands such as printBranches do not.

`runQuery` would imply to me that you are passing is a query - but you're actually generating the query in this function. And generateBranchSql is only called from here. So make that function be immediately follow this one to indicate they are strongly tied together.